code/modules/mining/machine_processing.dm

```diff
@@ -161,10 +222,62 @@
 	selected_material = getmaterialref(/datum/material/iron)
 
 /obj/machinery/mineral/processing_unit/Destroy()
+	if(CONSOLE)
+		SStgui.close_uis(CONSOLE)
 	CONSOLE = null
 	QDEL_NULL(stored_research)
 	return ..()
 
+/obj/machinery/mineral/processing_unit/RefreshParts()
+	var/point_upgrade_temp = 0
+	var/smelt_amount_temp = 1
+	for(var/obj/item/stock_parts/matter_bin/B in component_parts)
+		smelt_amount_temp += 1 + (1 * B.rating)
+	for(var/obj/item/stock_parts/micro_laser/L in component_parts)
+		point_upgrade_temp += 0.65 + (0.35 * L.rating)
+	point_upgrade = point_upgrade_temp
+	smelt_amount = round(smelt_amount_temp, 1)
+
+/obj/machinery/mineral/processing_unit/AltClick(mob/living/user)
+	if(!user.canUseTopic(src, BE_CLOSE))
+		return
+	if(panel_open)
+		input_dir = turn(input_dir, -90)
+		output_dir = turn(output_dir, -90)
+		to_chat(user, "<span class='notice'>You change [src]'s I/O settings, setting the input to [dir2text(input_dir)] and the output to [dir2text(output_dir)].</span>")
+		unregister_input_turf() // someone just rotated the input and output directions, unregister the old turf
+		register_input_turf() // register the new one
+		return TRUE
+
+/obj/machinery/mineral/processing_unit/attackby(obj/item/W, mob/user, params)
+	if(default_deconstruction_screwdriver(user, icon_state, icon_state, W))
+		return
+
+	if(default_unfasten_wrench(user, W))
+		return
+
+	if(default_deconstruction_crowbar(W))
+		return
+
+	if(W.tool_behaviour == TOOL_MULTITOOL)
+		if(!multitool_check_buffer(user, W))
+			return
+		var/obj/item/multitool/P = W
```

Comment:
This doesnt check for multitool tool behaviour, rather that the item is specifically of type multitool, so use a typecheck instead

Reply:
multitool_check_buffer performs a typecheck for /obj/item/multitool